src/pages/iou/request/step/withFullTransactionOrNotFound.js

```diff
@@ -0,0 +1,68 @@
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {withOnyx} from 'react-native-onyx';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import transactionPropTypes from '@components/transactionPropTypes';
+import getComponentDisplayName from '@libs/getComponentDisplayName';
+import ONYXKEYS from '@src/ONYXKEYS';
+import IOURequestStepRoutePropTypes from './IOURequestStepRoutePropTypes';
+
+const propTypes = {
+    /** The HOC takes an optional ref as a prop and passes it as a ref to the wrapped component.
+     * That way, if a ref is passed to a component wrapped in the HOC, the ref is a reference to the wrapped component, not the HOC. */
+    forwardedRef: PropTypes.func,
+
+    /** The report corresponding to the reportID in the route params */
+    transaction: transactionPropTypes,
+
+    route: IOURequestStepRoutePropTypes.isRequired,
+};
+
+const defaultProps = {
+    forwardedRef: () => {},
+    transaction: {},
+};
+
+export default function (WrappedComponent) {
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithFullTransactionOrNotFound({forwardedRef, ...props}) {
+        const {
+            transaction: {transactionID},
+        } = props;
+
+        // If the transaction does not have a transactionID, then the transaction no longer exists in Onyx as a full transaction and the not-found page should be shown
+        if (!transactionID) {
+            return <FullPageNotFoundView shouldShow />;
```

Comment:
This caused a regression https://github.com/Expensify/App/issues/32807, we should've shown the not found view when the component is focused.